db2i_charsetSupport.cc

```diff
@@ -759,15 +274,77 @@ int32 getConversion(enum_conversionDirection direction, const CHARSET_INFO* cs,
                 sizeof(hashKey))))
   {
     DBUG_PRINT("getConversion", ("Hash miss for direction=%d, cs=%s, ccsid=%d", direction, cs->name, db2CCSID));
-    rc= openNewConversion(direction, cs->csname, db2CCSID, &hashKey, conversion);
-    if (rc)
-      DBUG_RETURN(rc);
-  }
-  else
-  {
-    conversion= mapping->iconvDesc;
+
+    EncodingMapEntry *entry = nullptr;
+
+    for (auto& e : encoding_map) {
+      if (strcmp(e.csname, cs->csname) == 0) {
+        entry = &e;
+        break;
+      }
+    }
+
+    if (entry == nullptr) {
+      fprintf(stderr, "ibmdb2 error entry was null charset was not found\n");
+      getErrTxt(DB2I_ERR_UNSUPP_CHARSET, cs->csname);
+      DBUG_RETURN(DB2I_ERR_UNSUPP_CHARSET);
+    }
+
+    const char *iconv_db2 = entry->iconv_db2;
+    // IBM-12345 is the max length of temp buffer
+    char temp[10];
```

Comment:
```suggestion
    char temp[sizeof("IBM-12345")];
```
This makes it clear without the need for the comment